openhands/runtime/mcp/proxy/manager.py

```diff
@@ -89,9 +90,29 @@ async def mount_to_app(
         if not self.proxy:
             raise ValueError('FastMCP Proxy is not initialized')
 
+        def close_on_double_start(app):
+            async def wrapped(scope, receive, send):
+                start_sent = False
+
+                async def check_send(message):
+                    nonlocal start_sent
+                    if message['type'] == 'http.response.start':
+                        if start_sent:
+                            raise get_cancelled_exc_class()(
+                                'closed because of double http.response.start (mcp issue #883)'
```

Comment:
```suggestion
                                'closed because of double http.response.start (mcp issue https://github.com/modelcontextprotocol/python-sdk/issues/883)'
```